core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -624,96 +635,259 @@ public void addPhysicallyWrittenRows(long rows) {
+                if (0 != ff.mkdirs(detachedPath, mkDirMode)) {
+                    LOG.error()
+                            .$("cannot create detached partition folder [path=")
+                            .$(detachedPath)
+                            .$(']')
+                            .$();
+                    return StatusCode.PARTITION_DETACHED_FOLDER_CANNOT_CREATE;
+                }
+            }
+            setPathForPartition(detachedPath.trimTo(detachedRootLen), partitionBy, timestamp, false);
+            detachedPath.put(DETACHED_DIR_MARKER).$();
+            final int detachedPathLen = detachedPath.length();
+            if (ff.exists(detachedPath)) {
+                LOG.error()
+                        .$("detached partition folder already exist [path=")
+                        .$(detachedPath)
+                        .$(']')
+                        .$();
+                return StatusCode.PARTITION_ALREADY_DETACHED;
+            }
+
+            // rename partition folder to partition.detached
+            if (ff.rename(path, detachedPath) != Files.FILES_RENAME_OK) {
+                LOG.error()
+                        .$("cannot rename [errno=")
+                        .$(ff.errno())
+                        .$(", from=")
+                        .$(path)
+                        .$(", to=")
+                        .$(detachedPath)
+                        .$(']')
+                        .$();
+                return StatusCode.PARTITION_FOLDER_CANNOT_RENAME;
+            }
+
+            // copy _meta and _cv to partition.detached
+            other.of(path).trimTo(rootLen).concat(META_FILE_NAME).$(); // exists already checked
+            other2.of(detachedPath).trimTo(detachedPathLen).concat(META_FILE_NAME).$();
+            StatusCode statusCode = StatusCode.OK;
+            boolean copiedDetachedMeta = false;
+            if (-1 == ff.copy(other, other2)) {
+                statusCode = StatusCode.PARTITION_CANNOT_COPY_META;
+                LOG.error().$("cannot copy [errno=").$(ff.errno())
+                        .$(", from=").$(other)
+                        .$(", to=").$(other2)
+                        .I$();
+            } else {
+                copiedDetachedMeta = true;
+                other.parent().concat(COLUMN_VERSION_FILE_NAME).$();
+                other2.parent().concat(COLUMN_VERSION_FILE_NAME).$();
+                if (-1 == ff.copy(other, other2)) {
+                    statusCode = StatusCode.PARTITION_CANNOT_COPY_META;
+                    LOG.error().$("cannot copy [errno=").$(ff.errno())
+                            .$(", from=").$(other)
+                            .$(", to=").$(other2)
+                            .I$();
+                }
+            }
+            if (statusCode == StatusCode.OK) {
+                // all good, commit
+                for (int colIdx = 0; colIdx < columnCount; colIdx++) {
+                    columnVersionWriter.remove(timestamp, colIdx);
```

Comment:
Can you add method in `columnVersionWriter` to remove all column entries for the partition in one go?
Doing loop with remove from middle of list is not great.